lib/auth/auth.go

```diff
@@ -1319,6 +1319,11 @@ type newMFADeviceFields struct {
 	// u2fStorage is the storage used to hold the u2f challenge.
 	// Field can be empty to use default services.Identity storage.
 	u2fStorage u2f.RegistrationStorage
+
+	// webIdentityOverride is an optional RegistrationIdentity override to be used
+	// for device registration.
+	// Defaults to the Server's IdentityService.
+	webIdentityOverride wanlib.RegistrationIdentity
```

Comment:
When would we need this? What use case does the override allow?

Reply:
This so we can apply WithInMemorySessionData during registration. Added a comment.